Gini-iOS-SDK/GINIDocumentTaskManager.m

```diff
@@ -61,76 +61,127 @@ - (instancetype)initWithAPIManager:(GINIAPIManager *)apiManager {
+    return [self createDocumentWithFilename:fileName
+                                   fromData:data
+                                    docType:docType
+                          cancellationToken:nil];
 }
 
-- (BFTask *)createDocumentWithFilename:(NSString *)fileName fromImage:(UIImage *)image docType:(NSString *)docType cancellationToken:(BFCancellationToken *)cancellationToken {
-    return [self createDocumentWithFilename:fileName fromData:UIImageJPEGRepresentation(image, 0.2) docType:docType cancellationToken:cancellationToken];
-}
+- (BFTask *)createDocumentWithFilename:(NSString *)fileName
+                              fromData:(NSData *)data
+                               docType:(NSString *)docType
+                     cancellationToken:(BFCancellationToken *)cancellationToken {
+    NSParameterAssert([fileName isKindOfClass:[NSString class]]);
+    NSParameterAssert([data isKindOfClass:[NSData class]]);
+    
+    BFTask *createTask = [[_apiManager uploadDocumentWithData:data
+                                                  contentType:@"image/jpeg"
+                                                     fileName:fileName
+                                                      docType:docType
+                                            cancellationToken:cancellationToken] continueWithSuccessBlock:^id(BFTask *task) {
+        return [GINIDocument documentFromAPIResponse:task.result];
+    }];
+    return GINIhandleHTTPerrors(createTask);
 
-- (BFTask *)createDocumentWithFilename:(NSString *)fileName fromData:(NSData *)data docType:(NSString *)docType {
-    return [self createDocumentWithFilename:fileName fromData:data docType:docType cancellationToken:nil];
 }
 
-- (BFTask *)createDocumentWithFilename:(NSString *)fileName fromData:(NSData *)data docType:(NSString *)docType cancellationToken:(BFCancellationToken *)cancellationToken {
+- (BFTask *)createPartialDocumentWithFilename:(NSString *)fileName
+                                     fromData:(NSData *)data
+                                      docType:(NSString *)docType
+                            cancellationToken:(BFCancellationToken *)cancellationToken {
     NSParameterAssert([fileName isKindOfClass:[NSString class]]);
     NSParameterAssert([data isKindOfClass:[NSData class]]);
     
+    NSString* contentType = [data mimeType]; // i.e: image/jpeg
+    
+    NSString* lastContentTypeComponent = [[contentType componentsSeparatedByString:@"/"] lastObject];
+    NSString* concreteType = @"";  // i.e: jpeg
+    if (lastContentTypeComponent != nil && [lastContentTypeComponent length] > 0) {
+        concreteType = lastContentTypeComponent;
+    }
+    contentType = [NSString stringWithFormat:@"application/vnd.gini.v2.partial+%@", concreteType];
+    
     BFTask *createTask = [[_apiManager uploadDocumentWithData:data
-                                                  contentType:@"image/jpeg"
+                                                  contentType:contentType
```

Comment:
Here, you pass the document type and use it. But in the previous method it was hardcoded to jpeg.

Reply:
Answered above.